### DISCUSSION TITLE
Multiphase modelling without concentration parameter

### DISCUSSION THREAD

**Discussion by MScKarl** [idaholab/moose#24321] [Q&A General]
Hi everyone!

I have been using MOOSE for a while but there are still things I really can’t grasp. This is the only place I can turn to for help and I am in desperate need of some.

For some time, I have tried to convert my single-phase parameter algorithm into a dual-phase parameter algorithm. The problem I seem to face is that multiple of the required modules (switching functions, barrier functions, and penalty functions), require me to define a concentration of the different phases. In my case, I do not want to use the concentration.

The mathematical equations that I am using (which are shown in the images below) use mechanical stress to drive the phase transformation. How can I in the best possible way define this in MOOSE? The equations are already combined into one in f_eta, but when I run my code, everything changes phase, which isn’t right.

I would very much appreciate any help, explanations, and code-changing suggestions.

Best Regards
Karl

![equations](https://github.com/idaholab/moose/assets/98493448/b9831a83-6949-4d06-b801-dcb4cd6d0942)
```
[Mesh]
    type = GeneratedMesh
    dim = 2
    nx = 400
    ny = 400
    xmin = -3
    xmax = 3
    ymin = -3
    ymax = 3
    elem_type = QUAD4
[]

[AuxVariables]
  [./local_energy]
    order = CONSTANT
    family = MONOMIAL
  [../]

  [./cross_energy]
    order = CONSTANT
    family = MONOMIAL
  [../]
[]

[Variables]
  [./eta1]
    order = FIRST
    family = LAGRANGE
  [../]

  [./eta2]
    order = FIRST
    family = LAGRANGE
  [../]

[]

[ICs]
  [./eta1]
    variable = eta1
    type = RandomIC
    min = 1e-5
    max = 1e-3
  [../]

  [./eta2]
    variable = eta2
    type = RandomIC
    min = 1e-5
    max = 1e-3
  [../]
[]

[BCs]

   [./left_1]
    type = NeumannBC
    variable = 'eta1'
    boundary = 'bottom'
    value = 0.0
  [../]

  [./left_2]
    type = NeumannBC
    variable = 'eta2'
    boundary = 'bottom'
    value = 0.0
  [../]

   [./right_1]
    type = NeumannBC
    variable = 'eta1'
    boundary = 'bottom'
    value = 0.0
  [../]

  [./right_2]
    type = NeumannBC
    variable = 'eta2'
    boundary = 'bottom'
    value = 0.0
  [../]

   [./top_1]
    type = NeumannBC
    variable = 'eta1'
    boundary = 'bottom'
    value = 0.0
  [../]

  [./top_2]
    type = NeumannBC
    variable = 'eta2'
    boundary = 'bottom'
    value = 0.0
  [../]

  [./bottom_1]
    type = NeumannBC
    variable = 'eta1'
    boundary = 'bottom'
    value = 0.0
  [../]

  [./bottom_2]
    type = NeumannBC
    variable = 'eta2'
    boundary = 'bottom'
    value = 0.0
  [../]

[]

[AuxKernels]

  [./cross_terms]
    type = CrossTermGradientFreeEnergy
    variable = cross_energy
    interfacial_vars = 'eta1 eta2'

    kappa_names = 'kappa11 kappa12 
                   kappa21 kappa22' 
                   
  [../]

[]

[Kernels]
  #
  # Order parameter eta1
  #
  # Time derivative 1 
  [./deta1_dt]
    type = TimeDerivative
    variable = eta1
  [../]

  #Interface 1
  [./ACIntF_1]
    type = ACMultiInterface
    variable = eta1
    etas = 'eta1 eta2'
    mob_name = L1
    kappa_names = 'kappa11 kappa12' 
  [../]

  #Bulk 1
  [./ACbulk_eta1]
    type = AllenCahn
    variable = eta1
    coupled_variables = 'eta2'
    mob_name = L1
    f_name = F
  [../]

  #
  # Order parameter eta2
  #
  # Time derivative 2      
  [./deta2_dt]
    type = TimeDerivative
    variable = eta2
  [../]

  #Interface 2
  [./ACIntF_2]
    type = ACMultiInterface
    variable = eta2
    etas = 'eta1 eta2'
    mob_name = L2
    kappa_names = 'kappa21 kappa22' 
  [../]

  #Bulk 2
  [./ACbulk_eta2]
    type = AllenCahn
    variable = eta2
    coupled_variables = 'eta1'
    mob_name = L2
    f_name = F
  [../]
[]

[Materials]

  [./kappa]
    type = GenericConstantMaterial
    prop_names  = 'kappa11 kappa12 kappa21 kappa22'
    prop_values = '0.0000000133		0	0.000000334	0'
  [../]

   [./L1]
    type = GenericConstantMaterial
    prop_names = 'L1'
    prop_values = '1.0'
  [../]

   [./L2]
    type = GenericConstantMaterial
    prop_names = 'L2'
    prop_values = '1.0'
  [../]

  [./P0]
    type = GenericConstantMaterial
    prop_names = 'P0'
    prop_values = '144000000'
  [../]

  [./s]
    type = GenericConstantMaterial
    prop_names = 's'
    prop_values = '0.6'
  [../]

  [./f_eta]
    type = DerivativeParsedMaterial
    property_name = f_eta # Name of the function
    coupled_variables = 'eta1 eta2' 
    constant_names = 'P0 s'
    constant_expressions = '144*10^6 0.6'
    expression = 'P0*(((eta1^2-1)^2)+((eta2^2-1)^2)*(s/4*(eta1*(eta1+2)*(eta1^2+2*eta1-6)-7)))'
  [../]

[]

[Postprocessors]
  
  [./eta_1]
    type = ElementIntegralVariablePostprocessor
    variable = 'eta1'
  [../]
  [./eta_2]
    type = ElementIntegralVariablePostprocessor
    variable = 'eta2'
  [../]

[]


[Preconditioning]

    type = SMP
    full = true
  [../]

[]

[Executioner]
  type = Transient
  scheme = 'bdf2'

  solve_type = 'NEWTON'

  start_time = 0.0
  end_time   = 500.0

  [./TimeStepper]
    type = SolutionTimeAdaptiveDT
    dt = 0.1
  [../]
[]

[Debug]
  # show_var_residual_norms = true
[]

[Outputs]

  [./exodus]
    type = Exodus
  [../]

[]

```
Best regards
Karl

**Comment by MScKarl** [comment]
Hi @dschwen @amjokisaari!

In your profile, which I got from the MOOSE discussion page, it said that you are working in computational material science, which is the same area in which I tried to apply MOOSE. I was wondering if you maybe could have a look at my question. Maybe you could give me some suggestions or input on how I should proceed. If anything is unclear or if there is something I need to specify please let me know.

Best regards
Karl

**Comment by MScKarl** [comment]
@GiudGiud Maybe you are able to help me? I am in urgent need of help.

Best regards
Karl

**Reply by GiudGiud** [reply]
I'm sorry this (phase field and mechanics) is not my area of expertise. 
@laagesen or @hugary1995 will know more than me.

**Reply by MScKarl** [reply]
Hi @GiudGiud! 
I understand, but thanks for your reply. Is there a quick way to contact these people, as I'm a bit pressed for time?

**Reply by GiudGiud** [reply]
Their emails are public you can try that. 
They are both active on this forum they'll pitch in if they have time.

**Reply by hugary1995** [reply]
I don't have expertise on this matter, sorry. But before others chime in, I have the following suggestions/questions for starters:
1. remove the Neumann BCs, they are unnecessary unless you want to prescribe nonzero fluxes.
2. You said "The mathematical equations that I am using (which are shown in the images below) use mechanical stress to drive the phase transformation." but I don't see any elastic energy contribution in your free energy.

**Reply by laagesen** [reply]
To add to what @hugary1995 said, I'm not sure what the evolution equations of the model are. Do you have Allen-Cahn equations for eta1 and eta2 derived from this? There would need to be gradient energy terms defined, as well as elastic energy terms as @hugary1995 mentioned.

**Reply by MScKarl** [reply]
Hi!
Thank you for your answers. In the images below is the Allen-Cahn equation used and the Total free energy equation that is being used. 
![Allen-Cahn](https://github.com/idaholab/moose/assets/98493448/599eedb2-1cb7-4296-88ce-97fae7914003)
Гij is a diagonal matrix referred to as the mobility tensor and is multiplied by the functional derivative of the total free energy 𝐹
![Total free energy](https://github.com/idaholab/moose/assets/98493448/2a0041b6-b708-45a5-a797-19d93b39501e)

**Reply by hugary1995** [reply]
You can start from this example: modules/combined/examples/phase_field-mechanics/SimplePhaseTrans.i

**Reply by MScKarl** [reply]
Thank you. I will go through the example. I may ask some follow up questions later if I don't solve it. I hope that's okay?

**Reply by hugary1995** [reply]
Yes sure, just post it here. All of us try our best to offer free premium support within our bandwidth :)

**Comment by MScKarl** [comment]
Hi again!
I apologize that I return so quickly, but I just realized that I made a small mistake in my equations regarding the total free energy. 
What I missed was that the elastic energy is included in the potential and should not be handled separately. In the first-mentioned equations, all energy contributions are mathematically included except the gradient term (i.e. the interface contribution). This means that the mechanical part comes implicitly through its (x,y)- dependency. This would in turn mean that the mechanical energy is only spatial dependent and should not evolve with the phase evolution. The problem for me is to try to solve the equations:

![equations](https://github.com/idaholab/moose/assets/98493448/f15b2c5a-574b-403c-acce-b03dcc4cc75b)

**Reply by hugary1995** [reply]
I'm totally lost. Someone else please help...

**Reply by laagesen** [reply]
Based on the equations you've posted here, you should be able to solve the problem using two Allen-Cahn equations, one for `eta1` and another for `eta2`. For each one, you need 3 kernels: TimeDerivative, ACInterface, and AllenCahn. Based on the form of the gradient terms, you don't need to use ACMultiInterface. You can need to pass your free energy `f_eta` to each of the two Allen-Cahn equations. Please have a look at this documentation and study how each kernel implements the governing equation as I've described: https://mooseframework.inl.gov/modules/phase_field/Phase_Field_Equations.html
